include/networkit/graph/Graph.hpp

```diff
@@ -640,6 +640,18 @@ class Graph final {
         const node u;
     };
 
+    /**
+     * Removes edges adjacent to a node according to a specific criterion.
+     *
+     * @param u The node whose adjacent edges shall be removed.
+     * @param condition A function that takes a node as an input and returns a
+     * bool. If true the edge (u, v) is removed.
+     * @param edgesIn Whether in-going or out-going edges shall be removed.
+     * @return count The number of removed edges.
+     */
+    template <typename Condition>
```

Comment:
Implementation is not available to user-code as provided in `Graph.cpp` file.

Reply:
Now it is public.

Reply:
But template-instatiation does not work for user code, as the implementation is in a different compilation uint (`Graph.cpp`); if you have a templated method, the implementation has to be provided in the header file (e.g., `forNodes`).

Reply:
True, my bad I forgot that. I moved the method to the header.